query_base.go

```diff
@@ -1096,3 +1096,40 @@ func (q cascadeQuery) appendCascade(fmter schema.Formatter, b []byte) []byte {
 	}
 	return b
 }
+
+//------------------------------------------------------------------------------
+
+type idxHintsQuery struct {
+	idxHints [][]schema.QueryWithArgs
```

Comment:
Can't we flatten this `[]schema.QueryWithArgs`? Not sure if we need to support multiple `USE INDEX` clauses...

Reply:
Yeah, I agree
But here is https://dev.mysql.com/doc/refman/5.7/en/index-hints.html has interesting example:

```sql
SELECT * FROM t1 USE INDEX (i1) USE INDEX (i1,i1);
```

Reply:
The doc says that it is not an error, but it also does not say that it is useful... is it useful?

Reply:
Yeah, you are right